packaging/os/yumrepo.py

```diff
@@ -0,0 +1,529 @@
+                # This is here to prevent the checkmod param to appear in the
+                # dump output (Dunno how it appears here as it's filtered out
+                # in the add() method. Please help!).
+                if key not in self.hidden_params:
+                    repo_string += "%s = %s\n" % (key, value)
+
+            repo_string += "\n"
+
+        return repo_string
+
+
+def main():
+    # Module settings
+    module = AnsibleModule(
+        argument_spec=dict(
+            bandwidth=dict(),
+            baseurl=dict(),
+            cost=dict(),
+            description=dict(),
+            enabled=dict(type='bool'),
+            enablegroups=dict(type='bool'),
+            exclude=dict(),
+            failovermethod=dict(choices=['roundrobin', 'priority']),
+            file=dict(),
+            gpgcakey=dict(),
+            gpgcheck=dict(type='bool'),
+            gpgkey=dict(),
+            http_caching=dict(choices=['all', 'packages', 'none']),
+            includepkgs=dict(),
+            keepalive=dict(type='bool'),
+            metadata_expire=dict(),
+            metalink=dict(),
+            mirrorlist=dict(),
+            mirrorlist_expire=dict(),
+            name=dict(required=True),
+            password=dict(no_log=True),
+            protect=dict(type='bool'),
+            proxy=dict(),
+            proxy_password=dict(no_log=True),
+            proxy_username=dict(),
+            repo_gpgcheck=dict(type='bool'),
+            reposdir=dict(default='/etc/yum.repos.d'),
+            retries=dict(),
+            skip_if_unavailable=dict(type='bool'),
+            sslcacert=dict(),
+            ssl_check_cert_permissions=dict(type='bool'),
+            sslclientcert=dict(),
+            sslclientkey=dict(),
+            sslverify=dict(type='bool'),
+            state=dict(choices=['present', 'absent'], default='present'),
+            throttle=dict(),
+            timeout=dict(),
+            username=dict(),
+        ),
+        add_file_common_args=True,
+        supports_check_mode=True,
+    )
+
+    name = module.params['name']
+    state = module.params['state']
+
+    # Rename "name" and "description" to ensure correct key sorting
+    module.params['repoid'] = module.params['name']
+    module.params['name'] = module.params['description']
+    del module.params['description']
+
+    # Instantiate the YumRepo object
+    yumrepo = YumRepo(module)
+
+    # Get repo status before change
+    yumrepo_before = yumrepo.dump()
+
+    # Perform action depending on the state
+    if state == 'present':
+        yumrepo.add()
+    elif state == 'absent':
+        yumrepo.remove()
+
+    # Get repo status after change
+    yumrepo_after = yumrepo.dump()
+
+    # Compare repo states
+    changed = yumrepo_before != yumrepo_after
+
+    # Save the file only if not in check mode and if there was a change
+    if not module.check_mode and changed:
+        yumrepo.save()
```

Comment:
This could raise (because of file permissions/missing directories/out of disk space/whatever).  AFAIU the [module guidelines](http://docs.ansible.com/ansible/developing_modules.html#module-checklist) ask that you catch the IOError here and replace it with a `module.fail_json(...)` call.